src/app/reducers/comment.reducer.spec.ts

```diff
@@ -0,0 +1,266 @@
+        parentId: null,
+        selfLink: '/1'
+      }
+    ];
+    const action = new CommentActions.GetSuccess(comments);
+    const state = CommentReducer(CommentInitialState, action);
+
+    expect(state).toEqual(comments);
+  });
+
+  it('GetError Actions should return previous state', () => {
+    const comments: CommentUI[] = [
+      {
+        id: '1',
+        body: 'comment 1',
+        markup: 'MarkUp',
+        createdAt: '00:00',
+        creatorId: '1',
+        bodyRendered: '<p>comment 1</p>',
+        parentId: null,
+        selfLink: '/'
+      }
+    ];
+    const action = new CommentActions.GetError();
+    const state = CommentReducer(comments, action);
+
+    expect(comments).toEqual(state);
+  });
+
+  it('AddSuccess actions should retrun the updated state', () => {
+    const comments: CommentUI[] = [
+      {
+        id: '1',
+        body: 'comment 1',
+        markup: 'MarkUp',
+        createdAt: '00:00',
+        creatorId: '1',
+        bodyRendered: '<p>comment 1</p>',
+        parentId: null,
+        selfLink: '/'
+      }
+    ];
+    const newComment: CommentUI = {
+      id: '2',
+      body: 'comment 2',
+      markup: 'MarkUp',
+      createdAt: '00:00',
+      creatorId: '1',
+      bodyRendered: '<p>comment 2</p>',
+      parentId: null,
+      selfLink: '/'
+    };
+
+    const newCommentState: CommentUI[] = [
+      {
+        id: '2',
+        body: 'comment 2',
+        markup: 'MarkUp',
+        createdAt: '00:00',
+        creatorId: '1',
+        bodyRendered: '<p>comment 2</p>',
+        parentId: null,
+        selfLink: '/'
+      },
+      {
+        id: '1',
+        body: 'comment 1',
+        markup: 'MarkUp',
+        createdAt: '00:00',
+        creatorId: '1',
+        bodyRendered: '<p>comment 1</p>',
+        parentId: null,
+        selfLink: '/'
+      }
+    ];
+
+    const getSuccessAction = new CommentActions.GetSuccess(comments);
+    const state = CommentReducer(CommentInitialState, getSuccessAction);
+
+    const addSuccessAction = new CommentActions.AddSuccess(newComment);
+    const newState = CommentReducer(state, addSuccessAction);
+
+    expect(newState).toEqual(newCommentState);
+  });
+
+  it('AddError action should return the previous state', () => {
+    const comments: CommentUI[] = [
+      {
+        id: '1',
+        body: 'comment 1',
+        markup: 'MarkUp',
+        createdAt: '00:00',
+        creatorId: '1',
+        bodyRendered: '<p>comment 1</p>',
+        parentId: null,
+        selfLink: '/'
+      }
+    ];
+
+    const getSuccessAction = new CommentActions.GetSuccess(comments);
+    const state = CommentReducer(CommentInitialState, getSuccessAction);
+
+    const addErrorAction = new CommentActions.AddError();
+    const newState = CommentReducer(state, addErrorAction);
+
+    expect(newState).toEqual(state);
+  });
+
+  it('updateSuccess action should return updates state', () => {
+    const comments: CommentUI[] = [
```

Comment:
You are using `const comments` in most of the specs. It would be better to move it to `beforeEach`. WDYT?

Reply:
Yes, I will move it to beforeEach.